.github/workflows/nightly.yml

```diff
@@ -0,0 +1,26 @@
+
+# This workflow includes more expensive tests than what is run on every PR, that are unlikely to break on most changes.
+# It also publishes a nightly release.
+name: Nightly test and release workflow
+
+on:
+  schedule:
+    # Chosen to be hopefully outside of business hours for most contributors'
+    # time zones, and not on the hour to avoid heavy scheduled-job times:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
+    - cron: "30 6 * * *"
+  workflow_dispatch:
+
+jobs:
+  deep-integration-tests:
+    uses: ./.github/workflows/integration-tests-reusable.yml
+    with:
+      all_platforms: true
+      num_shards: 5
+  publish-release:
+    uses: ./.github/workflows/publish-release-reusable.yml
+    with:
+      sha: ${{ refs/heads/master }}
```

Comment:
Not sure if this works, might have to do:

```
    steps:
      - id: get-sha
        run: |
          echo ::set-output name=sha::$( curl -u "u:${{github.token}}" https://api.github.com/repos/some-owner/some-repo/git/ref/heads/master | jq .object.sha | tr -d '"' )
      - uses: actions/checkout@v2
        with:
          ref: ${{ steps.get-sha.outputs.sha }}
```

Reply:
> You should be able to just use `${{ github.sha }}` as in https://github.com/dafny-lang/dafny/blob/master/.github/workflows/publish-release.yml#L16

How would that be defined for a cronjob? It's not a sha that's triggering the workflow.

Reply:
Ah right. :P